Tests/Engine/InvokeScriptAnalyzer.tests.ps1

```diff
@@ -630,5 +630,35 @@ Describe "Test -EnableExit Switch" {
                 $warnings.RuleName | Should -Be 'TypeNotFound'
             }
         }
+
+        Describe "Handles static Singleton (issue 1182)" {
+            $scriptDefinition = @'
+                enum LogLevel
+                {
+                    Error
+                    Information
+                    Verbose
+                    Debug
+                }
+
+                class Logger
+                {
+                    static [Logger] $instance
+                }
+
+                function Write-Log
+                {
+                    [CmdletBinding()]
+                    param(
+                        [LogLevel] $LogLevel
+                    )
+
+                    $logger.WriteLog($LogLevel, "Hello")
+                }
+'@
+            It "Does not throw or return diagnostic record" {
+                Invoke-ScriptAnalyzer -ScriptDefinition $scriptDefinition -ErrorAction Stop | Should -BeNullOrEmpty
```

Comment:
i think this should probably be { Invoke-ScriptAnalyzer -ScriptDefinition $scriptDefinition -ErrorAction Stop } | Should -Not Throw

Reply:
`-ErrorAction Stop` already takes care of making the test throw. Jacub who maintains Pester has said that there is no purpose for using `Should -Not Throw` because Pester will fail the test if the error is terminating (hence the `-ErrorAction Stop`